src/beautifiers/vhdl-beautifier/index.coffee

```diff
@@ -0,0 +1,59 @@
+###
+Requires https://www.gnu.org/software/emacs/
+###
+
+"use strict"
+Beautifier = require('../beautifier')
+path = require("path")
+
+module.exports = class VhdlBeautifier extends Beautifier
+  name: "Vhdl Beautifier"
+  link: "https://www.gnu.org/software/emacs/"
+  executables: [
+    {
+      name: "Emacs"
+      cmd: "emacs"
+      homepage: "https://www.gnu.org/software/emacs/"
+      installation: "https://www.gnu.org/software/emacs/"
+      version: {
+        parse: (text) -> text.match(/Emacs (\d+\.\d+\.\d+)/)[1]
+      }
+    }
+  ]
+
+  options: {
+    Vhdl: true
+  }
+
+  beautify: (text, language, options) ->
+    @debug('vhdl-beautifier', options)
+    emacs = @exe("emacs")
+
+    emacs_path = options.emacs_path
+    emacs_script_path = options.emacs_script_path
+
+    if not emacs_script_path
+      emacs_script_path = path.resolve(__dirname, "emacs-vhdl-formating-script.lisp")
+
+    @debug('vhdl-beautifier', 'emacs script path: ' + emacs_script_path)
+
+    args = [
+      '--batch'
+      '-l'
+      emacs_script_path
+      '-f'
+      'vhdl-batch-indent-region'
+      tempFile = @tempFile("temp", text)
+      ]
+
+    if emacs_path
+      @deprecateOptionForExecutable("Emacs", "emacs_path", "Path")
```

Comment:
This `emacs_path` option is no longer needed as use of Executables on line 12-20 replaces it.